dev-client/src/screens/CreateSiteScreen/components/CreateSiteForm.tsx

```diff
@@ -88,6 +88,14 @@ export const CreateSiteForm = ({
         <VStack p="16px" pt="30px" space="18px">
           <FormField name="name">
             <FormLabel>{t('site.create.name_label')}</FormLabel>
+
+            {/* TODO(performance):
+            1. All text inputs should be adjusted into uncontrolled components
+               - https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components
+               - https://github.com/facebook/react-native/issues/20119
+            2. The code should be rewritten/reorganized/memoized so that a change to a text input's value doesn't
+            trigger a re-render of the whole Form/Component (chokes the JS thread and results in terrible frame drops)
+            */}
```

Comment:
could you expand on how this would work? the form components aren't controlled by props when they're in a form, and are memoized. or are you saying that the entire architecture of Formik is causing the forms to re-render on each keystroke and we need to move away from the library?

Reply:
Sure, I'll try to elaborate on this further next week